File: rgctl-tests/ecommerce-cpp/src/cfg_features.cpp
// CFG feature probes for rgctl expected-facts (C++ lowering coverage).

namespace ecommerce {
namespace correctness {

int cfgShortCircuit(bool a, bool b) {
  if (a && b) {
    return 1;
  }
  return 0;
}

int cfgRangeFor() {
  int xs[] = {1, 2, 3};
  int total = 0;
  for (int v : xs) {
    total += v;
  }
  return total;
}

int cfgTryCatch(bool boom) {
  try {
    if (boom) {
      throw 1;
    }
    return 0;
  } catch (int) {
    return 1;
  }
}

int cfgIfInit(int n) {
  if (int x = n; x > 0) {
    return x;
  }
  return 0;
}

int cfgTernary(int x) { return x > 0 ? x : -x; }

}  // namespace correctness
}  // namespace ecommerce

File: rgctl-tests/ecommerce-javascript/src/cfg_features.js
/**
 * CFG feature probes for rgctl expected-facts (JavaScript lowering coverage).
 */

function cfgShortCircuit(a, b) {
  if (a && b) {
    return 1;
  }
  return 0;
}

function cfgForLoop(xs) {
  let total = 0;
  for (let i = 0; i < xs.length; i++) {
    total += xs[i];
  }
  return total;
}

function cfgTryCatch() {
  try {
    throw new Error("boom");
  } catch (e) {
    return 1;
  }
}

function cfgSwitch(x) {
  switch (x) {
    case 1:
      return 10;
    case 2:
      return 20;
    default:
      return 0;
  }
}

module.exports = { cfgShortCircuit, cfgForLoop, cfgTryCatch, cfgSwitch };
